spec/ref_arch_setup/bolt_helper_spec.rb

```diff
@@ -71,6 +72,48 @@
     end
   end
 
+  describe "run_task_with_bolt" do
+    before do
+      @env_vars = "VAR1=1 "
+      @env_vars << "VAR2=2"
+      @expected_command = "bolt task run #{task} VAR1=1 VAR2=2 --modulepath " +
+          "#{RefArchSetup::RAS_MODULE_PATH} --nodes #{nodes}"
+    end
+
+    context "when bolt works and returns true" do
+      it "returns true and outputs informative messages" do
+        expected_output = "All Good"
+        expected_status = 0
+        expect(RefArchSetup::BoltHelper).to receive(:`)\
+          .with(@expected_command).and_return(expected_output)
+        `(exit #{expected_status})`
+        expect($?).to receive(:success?).and_return(true) # rubocop:disable Style/SpecialGlobalVars
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Running: #{@expected_command}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts)\
+          .with("Exit status was: #{expected_status}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Output was: #{expected_output}")
+        expect(RefArchSetup::BoltHelper.run_task_with_bolt(task, @env_vars, nodes)).to eq(true)
+      end
+    end
+
+    context "when bolt fails and returns false" do
+      it "returns false and outputs informative messages and errors" do
+        expected_output = "No Good"
+        expected_status = 1
+        expect(RefArchSetup::BoltHelper).to receive(:`)\
+          .with(@expected_command).and_return(expected_output)
+        `(exit #{expected_status})`
+        expect($?).to receive(:success?).and_return(false) # rubocop:disable Style/SpecialGlobalVars
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Running: #{@expected_command}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("ERROR: bolt task failed!")
+        expect(RefArchSetup::BoltHelper).to receive(:puts)\
+          .with("Exit status was: #{expected_status}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Output was: #{expected_output}")
+        expect(RefArchSetup::BoltHelper.run_task_with_bolt(task, @env_vars, nodes)).to eq(false)
```

Comment:
if any of these expects fail it will be hard to read exactly why in rspec's output.  i suggest moving most, if not all expects to their own `it` clause (single expect, single test)